_dev/src/components/smart-shopping-campaign-creation/smart-shopping-campaign-creation.vue

```diff
@@ -678,41 +666,55 @@ export default {
-      this.campaignIsActive = this.foundSsc.status === CampaignStatus.ELIGIBLE;
-      [this.targetCountry] = this.$options.filters.changeCountriesCodesToNames([
-        this.foundSsc.targetCountry,
-      ]);
-      this.hasUnhandledFilters = this.foundSsc.hasUnhandledFilters;
-      this.debounceName();
-      this.$store.commit(
-        'smartShoppingCampaigns/SET_FILTERS_CHOSEN',
-        this.filtersChosen,
-      );
-      if (this.filtersChosen.length) {
-        let dimensionToEdit = this.$store.state.smartShoppingCampaigns.sscAvailableFilters.find(
-          (dim) => dim.id === this.filtersChosen[0].dimension,
-        );
-        const filtersToFind = this.filtersChosen[0].values;
-        dimensionToEdit = findAndCheckFilter(dimensionToEdit, filtersToFind);
-        this.$store.commit(
-          'smartShoppingCampaigns/SET_DIMENSION_CHOSEN',
-          dimensionToEdit,
-        );
-      }
+      this.$store
+        .dispatch('smartShoppingCampaigns/GET_DIMENSIONS_FILTERS', '')
+        .then(() => {
+          let {endDate} = this.foundSsc;
+          const todayYear = new Date().getFullYear();
 
-      this.loader = false;
+          if (new Date(endDate).getFullYear() - 10 > todayYear) {
+            endDate = null;
+          }
+          this.campaignName = this.foundSsc.campaignName;
+          this.campaignDurationStartDate = this.foundSsc.startDate;
+          this.campaignDurationEndDate = endDate;
+          this.campaignHasNoProductsFilter = !this.foundSsc.productFilters.length
+            && !this.foundSsc.hasUnhandledFilters;
+          this.campaignDailyBudget = this.foundSsc.dailyBudget;
+          this.campaignIsActive = this.foundSsc.status === CampaignStatus.ELIGIBLE;
+          [this.targetCountry] = this.$options.filters.changeCountriesCodesToNames([
+            this.foundSsc.targetCountry,
+          ]);
+          this.hasUnhandledFilters = this.foundSsc.hasUnhandledFilters;
+          this.debounceName();
+          this.$store.commit(
+            'smartShoppingCampaigns/SET_FILTERS_CHOSEN',
+            this.filtersChosen,
+          );
+          if (
+            this.filtersChosen.length
+            && this.$store.state.smartShoppingCampaigns.sscAvailableFilters.length
+          ) {
+            let dimensionToEdit = this.$store.state.smartShoppingCampaigns.sscAvailableFilters.find(
+              (dim) => dim.id === this.filtersChosen[0].dimension,
+            );
+            dimensionToEdit = findAndCheckFilter(
+              dimensionToEdit,
+              this.filtersChosen[0].values,
+            );
+            this.$store.commit(
+              'smartShoppingCampaigns/SET_DIMENSION_CHOSEN',
+              dimensionToEdit,
+            );
+            this.totalProducts = retrieveProductNumberFromFiltersIds(
+              this.filtersChosen,
+              this.$store.state.smartShoppingCampaigns.sscAvailableFilters,
+            );
```

Comment:
What about having a computed for this `totalProduct`?

Reply:
Not sure to agree there because `totalProducts` is a variable that is set / changed at different steps of the configuration, either set in edition mode, or when we change the filters that we chose, or when we change the dimension ... and the way it's modified is never the same

Reply:
All right, it seems there is room for improvement for this variable in the whole project. To be done in another PR